components/s3/setup.ftl

```diff
@@ -0,0 +1,135 @@
+                    ),
+                    {}
+                )
+            )
+        ]
+    [#else]
+        [#local encryptionConfiguration = {}]
+    [/#if]
+
+    [#-- Add NetworkACL Configuration --]
+    [#local virtualNetworkRulesConfiguration = []]
+    [#list solution.Access.SubnetIds as subnet]
+        [#local virtualNetworkRulesConfiguration += getStorageNetworkAclsVirtualNetworkRules(
+            id=subnet
+            action="Allow"
+        )]
+    [/#list]
+    [#local ipRulesConfiguration = []]
+    [#list solution.Access.IPAddressRanges as ip]
+        [#local ipRulesConfiguration += getStorageNetworkAclsIpRules(
+            value=ip
+            action="Allow"
+        )]
+    [/#list]
+    [#local networkAclsConfiguration = getStorageNetworkAcls(
+        defaultAction="Deny"
+        ipRules=ipRulesConfiguration
+        virtualNetworkRules=virtualNetworkRulesConfiguration
+        bypass="None"
+    )]
+
+    [#-- Add Container CORS Rules --]
+    [#if solution.CORSBehaviours]
+        [#local containerCorsRulesConfiguration = getStorageBlobServiceCorsRules(
+            solution.CORSBehaviours.AllowedOrigins
+            solution.CORSBehaviours.AllowedMethods
+            solution.CORSBehaviours.MaxAge
+            solution.CORSBehaviours.ExposedHeaders
+            solution.CORSBehaviours.AllowedHeaders
+        )]
+    [#else]
+        [#local containerCorsRulesConfiguration = {}]
+    [/#if]
+
+    [#if deploymentSubsetRequired("s3", true)]
+
+        [#-- TODO(rossmurr4y): Impliment tags. Currently the shared function getOccurrenceCoreTags
+        in gen3\engine\common.ftl just formats a call to the function getCfTemplateCoreTags, which is aws
+        provider specific. --]
+        [@createStorageAccount
+            name=accountId
+            kind=storageProfile.Type
+            sku=getStorageSku(storageProfile.Tier, storageProfile.Replication)
+            location=regionId
+            customDomain=
+                (solution.Website.CustomDomain)?has_content?then(
+                    getStorageCustomDomain(solution.Website.CustomDomain),
+                    {}
+                )
+            encryption=encryptionConfiguration
+            networkAcls=networkAclsConfiguration
+            accessTier=(storageProfile.AccessTier!{})
+            azureFilesIdentityBasedAuthentication=
+                (solution.Access.DirectoryService)?has_content?then(
+                    getStorageAzureFilesIdentityBasedAuthentication(solution.Access.DirectoryService),
+                    {}
+                )
+            supportsHttpsTrafficOnly=(solution.Website.HttpsOnly!"")
+            isHnsEnabled=(storageProfile.HnsEnabled!false)
+            dependsOn=dependencies
+        /]
+
+        [@createBlobService 
+            name=blobId
+            corsRules=containerCorsRulesConfiguration
+            deleteRetentionPolicy=
+                (solution.Lifecycle.BlobRetentionDays)?has_content?then(
```

Comment:
We have a common method for determining lifecycle rules which allow for overrides at the environment and product level. Might be worth looking at that for this as well

Reply:
Looking at the code for lifecycle rules shows its not  as generic as I thought. Lets leave this one and wait for https://github.com/codeontap/gen3/issues/998